alacritty_terminal/src/grid/mod.rs

```diff
@@ -73,6 +73,32 @@ pub trait GridCell {
 }
 
 /// Represents the terminal display contents
+///
+/// ```notrust
+/// ┌─────────────────────────┐  <-- max_scroll_limit + lines
+/// │                         │
+/// │      UNINITIALIZED      │
+/// │                         │
+/// ├─────────────────────────┤  <-- raw.len()
+/// │                         │
+/// │      RESIZE BUFFER      │
+/// │                         │
+/// ├─────────────────────────┤  <-- scroll_limit + lines
+/// │                         │
+/// │     SCROLLUP REGION     │
+/// │                         │
+/// ├─────────────────────────┤v lines
```

Comment:
```suggestion
/// ├─────────────────────────┤v <-- lines
```

Reply:
I've explicitly omitted the arrows there, since `lines` is not at that point, but it's the range of visible lines. You might want to compile the docs to see it better.